lgc/util/TypeLowering.cpp

```diff
@@ -91,6 +91,56 @@ TypeLowering::TypeLowering(LLVMContext &context) : m_builder(context) {
   addConstantRule(&coreConstantConverter);
 }
 
+// =====================================================================================================================
+// Lower function argument type based on the registered rules. If there is no type remapping needed, will just return
+// the old function, otherwise it will move all the instructions in the old function to the new function and return the
+// new function. So don't operate on the old function if new function was returned! The old function will be cleanup at
+// the time of TypeLowering::finishCleanup().
+//
+Function *TypeLowering::lowerFunctionArguments(Function &fn) {
+  SmallVector<Type *> newArgTys;
+  SmallVector<unsigned> remappedArgs;
+  for (size_t argIdx = 0; argIdx < fn.arg_size(); ++argIdx) {
+    auto *arg = fn.getArg(argIdx);
+    auto converted = convertType(arg->getType());
+    if (converted.size() == 1 && converted[0] == arg->getType()) {
```

Comment:
Can't you just exit the iteration if `converted.size() != 1`?

Reply:
I have shuffled the code and assert that `converted.size() == 1`.